spec/lib/upload_creator_spec.rb

```diff
@@ -43,6 +43,27 @@
         expect(File.extname(upload.url)).to eq('.png')
         expect(upload.original_filename).to eq('png_as.png')
       end
+
+      describe 'for webp format' do
+        before do
+          SiteSetting.authorized_extensions = 'webp|png'
+        end
+
+        let(:filename) { "webp_as.png" }
+        let(:file) { file_from_fixtures(filename) }
+
+        it 'should not correct the coerce filename' do
```

Comment:
@SamSaffron Is that what you meant?

Reply:
actually the particular case I mean we should test... take a legit `webp` file, rename it to file.bin. Allow `bin` uploads, upload the `bin` ... it should still be a `.bin`.

This is not the case for png/jpeg cause we would correct the filename.